internal/rules/register.go

```diff
@@ -113,6 +116,31 @@ func (r *registry) getFrameworkRules(fw ...framework.Framework) []RegisteredRule
 	return registered
 }
 
+func (r *registry) getSpecRules(spec string) []RegisteredRule {
+	r.RLock()
+	defer r.RUnlock()
+	var specRules []RegisteredRule
+
+	var complianceSpec cs.ComplianceSpec
+	specContent := specs.GetSpec(spec)
+	_ = yaml.Unmarshal([]byte(specContent), &complianceSpec)
```

Comment:
error here is ignored ?

Reply:
that's right - I ignored it for a couple of reasons:

1. We're unmarshaling something we've embedded ourselves. (no user input)
2. I wasn't sure what we should do if we encounter an error anyway. We could early return an empty slice back. 

What do you think?

Reply:
I think we should handle error in some way even if we put the json by our self as we also can make a mistake.
anyone might want to use this method in the future should know its safe to use.

in case of not handle error we could get nil pointer at least we should protect from nil pointer :
```for _, csRule := range complianceSpec.Spec.Controls {```

Reply:
Thanks good point, I added it here https://github.com/aquasecurity/defsec/pull/1045/commits/be774ff1f1025cb560ceffd3bc8c1264e473d418

Also another test should be able to cover the case of bogus spec from a loading point of view https://github.com/aquasecurity/defsec/pull/1045/files#diff-31f3b00f062e6251cb5439c6086f5e5a745d7caefda6db781101450933a2b354R19